packages/base/room.gts

```diff
@@ -246,6 +247,14 @@ class PatchField extends FieldDef {
   @field payload = contains(PatchObjectField);
 }
 
+// A map from card document hash to the first card fragment event id.
+// This map can be used to avoid sending the same version of the card more than once in a conversation.
+// We can reuse exisiting eventId if user attached the same version of the card.
+const cardDocToEventId: Map<string, string> = new Map();
+export function retrieveEventId(cardDoc: LooseSingleCardDocument) {
```

Comment:
let's rename this--it's _not_ an event Id--that's super confusing. I was quite confused when I first saw it. let's just call it a `cardHash`